src/tests/src/solver/optimisation/constraints/constraints_builder.cpp

```diff
@@ -0,0 +1,186 @@
+ * Adequacy and Performance assessment for interconnected energy networks.
+ *
+ * Antares_Simulator is free software: you can redistribute it and/or modify
+ * it under the terms of the Mozilla Public Licence 2.0 as published by
+ * the Mozilla Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * Antares_Simulator is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Mozilla Public Licence 2.0 for more details.
+ *
+ * You should have received a copy of the Mozilla Public Licence 2.0
+ * along with Antares_Simulator. If not, see <https://opensource.org/license/mpl-2-0/>.
+ */
+
+#define BOOST_TEST_MODULE "constraints_builder"
+#define WIN32_LEAN_AND_MEAN
+
+
+#include <numeric>
+#include <boost/test/unit_test.hpp>
+
+#include "antares/solver/optimisation/constraints/ShortTermStorageCumulation.h"
+
+#include "antares/solver/simulation/sim_structure_probleme_economique.h"
+#include "antares/antares/constants.h"
+
+
+
+struct BB
+{
+    int nombreDePasDeTempsPourUneOptimisation = 10;
+
+     std::vector <double>  Pi  = std::vector(2*nombreDePasDeTempsPourUneOptimisation, 0.0); // Placeholder for coefficients
+     std::vector<int>  Colonne = std::vector (2*nombreDePasDeTempsPourUneOptimisation, 0); // Placeholder for column indices
+     int nombreDeContraintes = 0;
+     int nombreDeTermesDansLaMatriceDeContrainte = 0;
+     std::vector<int>  IndicesDebutDeLigne = std::vector(4, 0);
+     SparseVector<double>   CoefficientsDeLaMatriceDesContraintes; //= std::vector(100, 0.0); // Plain vector
+     SparseVector<int>   IndicesColonnes; //= std::vector<int> (100, 0); // Plain vector for column indices
+     std::vector<int>  NombreDeTermesDesLignes = std::vector<int>(4, 0);
+      std::string Sens = std::string(4, '='); // Placeholder for constraint senses
+     int IncrementDAllocationMatriceDesContraintes = 10;
+     std::vector<CORRESPONDANCES_DES_VARIABLES> CorrespondanceVarNativesVarOptim;;
+    void set_correspondances_des_variables()
+    {
+        CorrespondanceVarNativesVarOptim.resize(nombreDePasDeTempsPourUneOptimisation);
+        for (auto i = 0; i < nombreDePasDeTempsPourUneOptimisation; i++)
+        {
+            CorrespondanceVarNativesVarOptim[i].SIM_ShortTermStorage = {.InjectionVariable = {0,1}, .WithdrawalVariable = {2,3} };
+        }
+    }
+
+    //.S {{.SIM_ShortTermStorage = {.InjectionVariable = {0,1}, .WithdrawalVariable = {2,3} } }};
+     const int32_t NombreDePasDeTempsPourUneOptimisation = nombreDePasDeTempsPourUneOptimisation; // Example value
+     std::vector<int>  NumeroDeVariableStockFinal = std::vector<int>(10, -1);
+     std::vector<std::vector<int> >  NumeroDeVariableDeTrancheDeStock = std::vector<std::vector<int>> (10, std::vector<int>(5, -1));
+     std::vector<std::string>  NomDesContraintes = std::vector<std::string>(100, "");
```

Comment:
Style
```suggestion
     std::vector<std::string>  NomDesContraintes(100);
```

Reply:
haha, doesn't work for in-class member initialization

Reply:
Sorry, this should work
```cpp
     std::vector<std::string>  NomDesContraintes{100};
```

Reply:
neither